tests/zfs-tests/tests/functional/cli_root/zfs_rollback/zfs_rollback_common.kshlib

```diff
@@ -116,7 +116,7 @@ function setup_snap_env
 			if datasetnonexists $snap; then
 				log_must cp /etc/passwd $fname
 				if is_linux || is_freebsd; then
-					log_must sync
+					sync_all_pools
```

Comment:
@allanjude this change reliably breaks the `zfs_rollback_001_pos`test.  I believe we need to keep the original sync here because we have a non-ZFS filesystem layered on a zvol and we need to make sure to flush the buffer in that filesystem.  It looks like we might be able to drop some of the special case handling for FreeBSD a few line down with a `log_must sync; sync_all_pools;` but I'd suggest we leave that for another PR.

Reply:
yeah, I also found another test where we measure the size of the file vdevs on the non-ZFS filesystem, where a `sync` is still required.